apps/meteor/client/views/omnichannel/contactInfo/ContactInfo/ReviewContactModal.tsx

```diff
@@ -0,0 +1,156 @@
+import type { CustomFieldMetadata, ILivechatContact, Serialized } from '@rocket.chat/core-typings';
+import { Badge, Box, Field, FieldError, FieldGroup, FieldHint, FieldLabel, FieldRow, Select } from '@rocket.chat/fuselage';
+import type { TranslationKey } from '@rocket.chat/ui-contexts';
+import { useAtLeastOnePermission, useEndpoint, useToastMessageDispatch, useTranslation } from '@rocket.chat/ui-contexts';
+import { useQueryClient } from '@tanstack/react-query';
+import React from 'react';
+import { Controller, useForm } from 'react-hook-form';
+
+import GenericModal from '../../../../components/GenericModal';
+import { useHasLicenseModule } from '../../../../hooks/useHasLicenseModule';
+import { ContactManagerInput } from '../../additionalForms';
+import { useCustomFieldsMetadata } from '../../directory/hooks/useCustomFieldsMetadata';
+
+const fieldNameMap: { [key: string]: TranslationKey } = {
+	name: 'Name',
+	contactManager: 'Contact_Manager',
+};
+
+function mapConflicts(
+	contact: Serialized<ILivechatContact>,
+	metadata: CustomFieldMetadata[],
+): Record<string, { name: string; label: string; values: string[] }> {
+	const conflicts = contact.conflictingFields?.reduce(
+		(acc, current) => {
+			const fieldName = current.field === 'manager' ? 'contactManager' : current.field.replace('customFields.', '');
+
+			if (acc[fieldName]) {
+				acc[fieldName].values.push(current.value);
+			} else {
+				acc[fieldName] = {
+					name: fieldName,
+					label:
+						(current.field.startsWith('customFields.') && metadata.find(({ name }) => name === fieldName)?.label) ||
+						fieldNameMap[fieldName],
+					values: [current.value],
+				};
+			}
+			return acc;
+		},
+		{} as Record<string, { name: string; label: string; values: string[] }>,
```

Comment:
I think we could elaborate a bit more the type here. also I would love to see this into a separated file with unit tests